apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLogger.cs

```diff
@@ -128,20 +123,52 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             }
 
             // If we have format params and its more than just the original message add them.
+            Struct paramStruct = null;
```

Comment:
You can initialize here.

Reply:
I figured I only initialize it when it's necessary. This avoids allocating a `Struct` instance when no log format parameters are passed.

Reply:
I would be more incline to have cleaner code than to avoid the allocation of one Struct instance tbh.